src/software/backend/robot_status.cpp

```diff
@@ -0,0 +1,82 @@
+#include "software/backend/robot_status.h"
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "software/logger/logger.h"
+
+TbotsRobotMsg convertRobotStatusToTbotsRobotMsg(RobotStatus* robot_status)
+{
+    std::map<std::string, ErrorCode> errorStringToEnumMap = {
+        {"charge timeout", CHARGE_TIMEOUT},
+        {"wheel 0 motor hot", WHEEL_0_MOTOR_HOT},
+        {"wheel 1 motor hot", WHEEL_1_MOTOR_HOT},
+        {"wheel 2 motor hot", WHEEL_2_MOTOR_HOT},
+        {"wheel 3 motor hot", WHEEL_3_MOTOR_HOT},
+        {"dribbler motor hot", DRIBBLER_MOTOR_HOT},
+        {"wheel 0 encoder not commutating", WHEEL_0_ENCODER_NOT_COMMUTATING},
+        {"wheel 1 encoder not commutating", WHEEL_1_ENCODER_NOT_COMMUTATING},
+        {"wheel 2 encoder not commutating", WHEEL_2_ENCODER_NOT_COMMUTATING},
+        {"wheel 3 encoder not commutating", WHEEL_3_ENCODER_NOT_COMMUTATING},
+        {"wheel 0 Hall sensor stuck low", WHEEL_0_HALL_SENSOR_STUCK_LOW},
+        {"wheel 1 Hall sensor stuck low", WHEEL_1_HALL_SENSOR_STUCK_LOW},
+        {"wheel 2 Hall sensor stuck low", WHEEL_2_HALL_SENSOR_STUCK_LOW},
+        {"wheel 3 Hall sensor stuck low", WHEEL_3_HALL_SENSOR_STUCK_LOW},
+        {"dribbler Hall sensor stuck low", DRIBBLER_HALL_SENSOR_STUCK_LOW},
+        {"wheel 0 Hall sensor stuck high", WHEEL_0_HALL_SENSOR_STUCK_HIGH},
+        {"wheel 1 Hall sensor stuck high", WHEEL_1_HALL_SENSOR_STUCK_HIGH},
+        {"wheel 2 Hall sensor stuck high", WHEEL_2_HALL_SENSOR_STUCK_HIGH},
+        {"wheel 3 Hall sensor stuck high", WHEEL_3_HALL_SENSOR_STUCK_HIGH},
+        {"dribbler Hall sensor stuck high", DRIBBLER_HALL_SENSOR_STUCK_HIGH},
+    };
+
+    // Insufficient ChipperKickerStatus, DriveUnits, and NetworkStatus info
```

Comment:
I'm a bit confused about what this comment means?

Reply:
It means that although `TbotsRobotMsg` has fields for `ChipperKickerStatus`, `DriveUnits`, and `NetworkStatus`, there isn't enough information from `RobotStatus` to fill those in.

Reply:
Ah, I see. Generally if some function is producing another type I'd expect that type to be filled out (for lack of a better word). If the `RobotStatus` represents some logical encapsulation of information, perhaps we could change the proto to reflect this so that we could return a single more "filled out" type here?

@MathewMacDougall @jonathanlew @akhilveeraghanta - what do you guys think about this? I'm struggling a bit to really give concrete reasons why, but this feels like it could be bug prone.

Reply:
Hmmm yea it's a good point. I know we originally weren't too concerned about missing values because in theory sensor fusion / filters should be able to work with any missing information. But at the same time I agree that my first assumption would always be that the type is "fully filled out".

At least because it's `proto3` there's a good way to represent "no data" so we won't have bugs with sentinel values.

I don't have any good suggestions to fix this either because we fundamentally don't have that data, and this message is being designed more to suit the switch to wifi. Not that we can't provide some of the missing information, but I'm assuming we don't want to touch the old firmware at all to support that.

I also don't think it's appropriate to create a sub-msg that only contains fields that the old firmware can supply, because that also complicates things and everything is supposed to receive the same top-level message anyway.

So I _guess_ I'm ok with this because I can't think of any alternatives, and this will hopefully be able to be removed within the year.